src/doc/unstable-book/src/language-features/register-tool.md

```diff
@@ -0,0 +1,48 @@
+# `register_tool`
+
+The tracking issue for this feature is: [#66079]
+
+[#66079]: https://github.com/rust-lang/rust/issues/66079
+
+------------------------
+
+The `register_tool` language feature informs the compiler that attributes in your code are meant to be used with tools other than the compiler itself. This can be useful if your code has semantic meaning without the external tool, but enables additional features when the tool is present.
+
+`register_tool` also allows configuring lint levels for external tools.
+
+Tool attributes are only meant for ignorable attributes. If your code *changes* meaning when the attribute is present, it should not use a tool attribute (because it cannot be compiled with anything other than the external tool, and in a sense is a fork of the language).
+
+------------------------
+
+`#![register_tool(tool)]` is an attribute, and is only valid at the crate root.
+Attributes using the registered tool are checked for valid syntax, and lint attributes are checked to be in a valid format. However, the compiler cannot validate the semantics of the attribute, nor can it tell whether the configured lint is present in the external tool.
+
+Semantically, `clippy::*`, `rustdoc::*`, and `rustfmt::*` lints and attributes all behave as if `#![register_tool(clippy, rustdoc, rustfmt)]` were injected into the crate root, except that the `rustdoc` namespace can only be used for lints, not for attributes.
+When compiling with `-Z unstable-features`, `rustc::*` lints can also be used. Like `rustdoc`, the `rustc` namespace can only be used with lints, not attributes.
+
+The compiler will emit an error if it encounters a lint/attribute whose namespace isn't a registered tool.
+
+Tool names cannot be "nested"; `register_tool(foo::bar)` is an error.
```

Comment:
Nit: ... maybe "namespaced"? But this is fine too.

Reply:
well, formally these tool names are called "namespaces". so saying "a tool name cannot be namespaced" seems very confusing. "tool namespaces cannot be nested" is maybe a little better, would you prefer that?

Reply:
Right. Sure, that sounds good to me :)